### ISSUE TITLE
fail to compile: adding subset of tuple to set

### ISSUE THREAD

**Issue by talkaminker** [serge-sans-paille/pythran#1124]
the following fails to compile (with a bunch of c++ errors. version 0.9.0)
```
# pythran export test((int))
def test(x):
    s = set()
    s.add(x[:10])
    return s
```

while just adding x directly to the set s will compile
this is fine:
```
# pythran export test((int))
def test(x):
    s = set()
    s.add(x)
    return s
```
also using `s.add((x[0],x[1],x[2],x[3]))` will compile.

tested on 64 bit python 3.6 windows
this is probably a bug. right?

**Comment by serge-sans-paille** [comment]
In Pythran tuples can only interact with constant expression, which basically means you cannot take a dynamic slice of it. But in your case, it's a static slice (all elements are known) so we should be able to handle that, although it's clearly not supported at the moment. Making the slice explicit indeed solves the issue - it's very close to unrolling :-)

**Comment by talkaminker** [comment]
Thank you.
Can you explain how come this works? (or was it not supposed to work and I am just lucky in this example)?
this works:
```
# pythran export fff((int), int)
def fff(x, i):
    return x[:i] + x[:i]
```
```
>>> fff(tuple(range(10)),3)
>>> (0,1,2,0,1,2)
```
it seems the interaction with set is problematic here.
Great package btw!

**Comment by serge-sans-paille** [comment]
The code above work because we have a specialization for tuples when all the elements share the same type. In that case, the tuple behvaes roughly as an array and it's possible to slice it, to iterate over it etc.

Concerning the first case, I've dug a bit and the problem lies in the set builtin, I've got a patch for that, just validating locally before pushing it.

Thanks a lot for spotting that!

**Comment by serge-sans-paille** [comment]
#1127 should fix this issue.

**Comment by serge-sans-paille** [comment]
Fixed by #1127
